mgnify_pipelines_toolkit/utils/rename_contig/cli_rename.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2024-2025 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rename command implementation for CLI."""
+
+from pathlib import Path
+
+import click
+
+from .core import read_mapping_file, rename_fasta, write_mapping_file
+from .handlers import GenBankHandler, GFFHandler
+from .parsers import parse_header, parse_virify_header
+
+SUPPORTED_FASTA_EXTS = (".fasta", ".fa", ".fna", ".fasta.gz", ".fa.gz", ".fna.gz")
+SUPPORTED_GBK_EXTS = (".gbk", ".gb", ".genbank")
+SUPPORTED_GFF_EXTS = (".gff", ".gff3", ".gff.gz", ".gff3.gz")
+
+
+@click.command()
+@click.argument("input", nargs=-1, required=True, type=click.Path(exists=True))
+@click.option("-o", "--outdir", required=True, type=click.Path(), help="Output directory")
+@click.option("-p", "--prefix", default="contig", help="Prefix for new names (default: contig)")
+@click.option("-m", "--map", type=click.Path(), help="Output mapping file (TSV)")
+@click.option(
+    "--use-mapping",
+    type=click.Path(exists=True),
+    help="Use existing mapping file instead of generating new names",
+)
+@click.option("--from-col", default="original", help="Source column in mapping file (default: original)")
+@click.option("--to-col", default="renamed", help="Target column in mapping file (default: renamed)")
+@click.option(
+    "--parser",
+    type=click.Choice(["base", "virify"]),
+    default="base",
+    help="Header parser strategy (default: base). Use 'virify' for Virify metadata",
+)
+@click.option(
+    "--formatter",
+    type=click.Choice(["simple", "generic", "virify"]),
+    default="simple",
+    help="Metadata formatter strategy (default: simple). 'generic' preserves all metadata, 'virify' handles Virify IDs",
+)
+@click.option("--separate-by-size", is_flag=True, help="Separate output by contig size (_1kb, _5kb, _100kb)")
```

Comment:
does it do all three contig sizes by default or can you control this?

Reply:
Also if it's a flag what's the default value? (i don't know what click does if you don't supply a default haha)